### ISSUE TITLE
MSSQL RDS multi_az and snapshot failure

### ISSUE THREAD

**Issue by dscowan** [terraform-providers/terraform-provider-aws#5594]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

v0.11.7

### Affected Resource(s)

* aws v1.31.0

### Terraform Configuration Files

```hcl
# Copy-paste your Terraform configurations here - for large Terraform configs,
# please use a service like Dropbox and share a link to the ZIP file. For
# security, you can also encrypt the files using our GPG public key: https://keybase.io/hashicorp
```
This fails 
```
resource "aws_db_instance" "default" {
  allocated_storage         = 50
  storage_type              = "gp2"
  license_model             = "license-included"
  engine                    = "sqlserver-se"
  engine_version            = "12.00.5571.0.v1"
  instance_class            = "db.m4.xlarge"
  identifier                = "testing-mirror"
  username                  = "test"
  password                  = "test"
  snapshot_identifier       = "base_snapshot"
  option_group_name         = "default:sqlserver-se-12-00"
  backup_retention_period   = 3
  skip_final_snapshot       = "false"
  final_snapshot_identifier = "test-mssql-final-snapshot"
  apply_immediately         = true
  multi_az                  = true
}
```

This works
```
resource "aws_db_instance" "default" {
  allocated_storage         = 50
  storage_type              = "gp2"
  license_model             = "license-included"
  engine                    = "sqlserver-se"
  engine_version            = "12.00.5571.0.v1"
  instance_class            = "db.m4.xlarge"
  identifier                = "testing-mirror"
  username                  = "test"
  password                  = "test"
  snapshot_identifier       = "base_snapshot"
  option_group_name         = "default:sqlserver-se-12-00"
  backup_retention_period   = 3
  skip_final_snapshot       = "false"
  final_snapshot_identifier = "test-mssql-final-snapshot"
  apply_immediately         = true
  multi_az                  = false
}
```

This works
```
resource "aws_db_instance" "default" {
  allocated_storage         = 50
  storage_type              = "gp2"
  license_model             = "license-included"
  engine                    = "sqlserver-se"
  engine_version            = "12.00.5571.0.v1"
  instance_class            = "db.m4.xlarge"
  identifier                = "testing-mirror"
  username                  = "test"
  password                  = "test"
  #snapshot_identifier       = "base_snapshot"
  option_group_name         = "default:sqlserver-se-12-00"
  backup_retention_period   = 3
  skip_final_snapshot       = "false"
  final_snapshot_identifier = "test-mssql-final-snapshot"
  apply_immediately         = true
  multi_az                  = true
}
```

### Panic Output

rror: Error applying plan:

1 error(s) occurred:

* aws_db_instance.default: 1 error(s) occurred:

* aws_db_instance.default: Error creating DB Instance: InvalidParameterValue: Mirroring cannot be applied to instances with backup retention set to zero.
        status code: 400, request id: 593f7b1b-8a4d-4086-bf21-c904a82bf4d1

Terraform does not automatically rollback in the face of errors.
Instead, your Terraform state file has been partially updated with
any resources that successfully completed. Please address the error
above and apply again to incrementally change your infrastructure.
### Expected Behavior
I would expect to be able to supply a snapshot_identifier and multi_az.  I can apply from new with snapshot identifier removed and get multi-az.  I can apply without a snapshot identifier from new and get multi-az.  I can apply from new without multi-az and then change multi_az to true and apply with success.  multi_az = true and snapshot = "XXXXX" does not work.

### Actual Behavior
multi_az = true with snapshots should be supported

### Steps to Reproduce

1. `terraform apply`

### References

* #0000

**Comment by bflad** [comment]
Hi @dscowan 👋  Sorry you ran into trouble here.

This indeed was a bug with the provider in the scenario you described, specifically with SQL Server and attempting to enable mirroring directly from restoring a database snapshot. I have submitted a bug fix pull request here: #5613

**Comment by bflad** [comment]
The fix for this has been merged and will release with version 1.33.0 of the AWS provider, likely later this week. 👍

**Comment by bflad** [comment]
This has been released in [version 1.33.0 of the AWS provider](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#1330-august-22-2018). Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.
